orderbook/src/orderbook.rs

```diff
@@ -323,40 +301,20 @@ async fn filter_unsupported_tokens(
     Ok(orders)
 }
 
-// Make sure the balance fetcher tracks all balances for user, sell token combinations in these
-// orders and returns said balances. Only records this for open orders.
-async fn track_and_get_balances(
+async fn get_balances(
     fetcher: &dyn BalanceFetching,
     orders: &[Order],
 ) -> HashMap<(H160, H160, SellTokenSource), U256> {
     let mut balances = HashMap::<(H160, H160, SellTokenSource), U256>::new();
-    let mut untracked = HashSet::<(H160, H160, SellTokenSource)>::new();
     for order in orders {
         if !matches!(order.order_meta_data.status, OrderStatus::Open) {
             continue;
         }
-        let key = (
-            order.order_meta_data.owner,
-            order.order_creation.sell_token,
-            order.order_creation.sell_token_balance,
-        );
-        match fetcher.get_balance(key.0, key.1, key.2) {
-            Some(balance) => {
-                balances.insert(key, balance);
-            }
-            None => {
-                untracked.insert(key);
-            }
+        let query = crate::account_balances::Query::from_order(order);
+        if let Ok(balance) = fetcher.get_balances(&[query]).await[0] {
```

Comment:
Assuming this is just temporary, and eventually you will want to build a `Vec<Query>` for all eligible orders to retrieve balances for?

Reply:
Yes that happens in the next PR.